plugins/modules/apt_rpm.py

```diff
@@ -155,25 +210,51 @@ def main():
         argument_spec=dict(
             state=dict(type='str', default='present', choices=['absent', 'installed', 'present', 'removed']),
             update_cache=dict(type='bool', default=False),
-            package=dict(type='list', elements='str', required=True, aliases=['name', 'pkg']),
+            clean=dict(type='bool', default=False),
+            dist_upgrade=dict(type='bool', default=False),
+            update_kernel=dict(type='bool', default=False),
+            package=dict(type='list', elements='str', aliases=['name', 'pkg']),
         ),
     )
 
     if not os.path.exists(APT_PATH) or not os.path.exists(RPM_PATH):
         module.fail_json(msg="cannot find /usr/bin/apt-get and/or /usr/bin/rpm")
 
     p = module.params
+    modified = False
+    output = ""
 
     if p['update_cache']:
-        update_package_db(module)
+        (m, out) = update_package_db(module)
+        modified = modified or m
+        output += out
+
+    if p['clean']:
+        (m, out) = clean(module)
+        modified = modified or m
+
+    if p['dist_upgrade']:
+        (m, out) = dist_upgrade(module)
+        modified = modified or m
+        output += out
 
+    if p['update_kernel']:
+        (m, out) = update_kernel(module)
+        modified = modified or m
+        output += out
+
+    # If package is missing return exit_json
     packages = p['package']
 
-    if p['state'] in ['installed', 'present']:
-        install_packages(module, packages)
+    if not isinstance(packages, list):
+        module.exit_json(changed=modified, msg=output)
+
+    else:
+        if p['state'] in ['installed', 'present']:
+            install_packages(module, packages)
```

Comment:
I don't see how `modified` is passed on to `install_packages` or `remove_packages` so these functions can return the correct `changed` information.